### PR TITLE
docs: update readme for cross lan communicaiton feature

### PR DESCRIPTION
**Pull request by khalid-davis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,33 +128,45 @@ EdgeMesh satisfies the new requirements in edge scenarios (e.g., limited edge re
...
 
 
 ## Architecture
 
-<img src="images/em-arch.png" style="zoom:67%;" />
+![image](./images/em-arch.png)  
 
-To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation of the traffic proxy. In addition, it also comes with a lightweight DNS resolver. As shown in the figure above, the core components of EdgeMesh include:
+To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation of the traffic proxy. In addition, it also comes with a lightweight DNS resolver.
+As shown in the figure above, EdgeMesh contains EdgeMesh-Server microservices and EdgeMesh-Agent microservices.
```
Comment (Poorunga) [inline README.md:142]: microservices or components? I think component  may be more appropriate

File: README.md [hunk 2]
```diff
@@ -166,7 +178,22 @@ Before using EdgeMesh, you need to understand the following prerequisites at fir
 - while using DestinationRule, the name of the DestinationRule must be equal to the name of the corresponding Service. Edgemesh will determine the DestinationRule in the same namespace according to the name of the Service
 - Service ports must be named. The key/value pairs of port name must have the following syntax: name: \<protocol>[-\<suffix>]
 
+#### Download Edgemesh
+```Shell
+git clone https://github.com/khalid-davis/edgemesh.git
```
Comment (Poorunga) [inline README.md:183]: Why did you replace the repo?

**Review by Poorunga — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (khalid-davis) [reply README.md:142]: EdgeMesh-Server and EdgeMesh-Agent are microservices of EdgeMesh Project, and both of them include some componnets

File: README.md (hunk 2 above)
Comment (khalid-davis) [reply README.md]: fix

File: README.md (hunk 1 above)
Comment (ZBoIsHere) [reply README.md:142]: I think component might be better

File: README.md [hunk 3]
```diff
@@ -128,33 +128,45 @@ EdgeMesh satisfies the new requirements in edge scenarios (e.g., limited edge re
...
 ## Architecture
 
-<img src="images/em-arch.png" style="zoom:67%;" />
+![image](./images/em-arch.png)  
 
-To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation of the traffic proxy. In addition, it also comes with a lightweight DNS resolver. As shown in the figure above, the core components of EdgeMesh include:
+To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation of the traffic proxy. In addition, it also comes with a lightweight DNS resolver.
+As shown in the figure above, EdgeMesh contains EdgeMesh-Server microservices and EdgeMesh-Agent microservices.  
+The core components of EdgeMesh-Server include:
+- **Tunnel**: Based on [libp2p](https://github.com/libp2p/go-libp2p), establish a connection with EdgeMesh-Agent to provide relay capability and hole punching capability
```
Comment (ZBoIsHere) [inline README.md:144]: I think Tunnel-Server might be better

File: README.md [hunk 4]
```diff
@@ -128,33 +128,45 @@ EdgeMesh satisfies the new requirements in edge scenarios (e.g., limited edge re
...
 
+The core components of EdgeMesh-Agent include: 
 - **Proxier**: Responsible for configuring the kernel's iptables rules, and intercepting requests to the EdgeMesh process
 - **DNS**: Built-in DNS resolver, which resolves the DNS request in the node into a service cluster IP
 - **Traffic**: A traffic forwarding module based on the Go-chassis framework, which is responsible for forwarding traffic between applications
 - **Controller**: Obtains metadata (e.g., Service, Endpoints, Pod, etc.) through the list-watch capability on the edge side of KubeEdge
-
+- **Tunnel**: Based on [libp2p](https://github.com/libp2p/go-libp2p), using relay and hole punching to provide the ability of communicating across subnets
```
Comment (ZBoIsHere) [inline README.md:152]: I think Tunnel-Agent might be better

File: README_zh.md [hunk 5]
```diff
@@ -225,7 +255,6 @@ daemonset.apps/edgemesh-agent created
 ```
 
 
-
 #### 测试样例
 
 **HTTP协议**
```
Comment (ZBoIsHere) [inline README_zh.md:260]: `kubectl apply -f example/hostname.yaml` can be modified to `kubectl apply -f examples/hostname.yaml`

File: README.md (hunk 3 above)
Comment (khalid-davis) [reply README.md]: fix

File: README.md (hunk 4 above)
Comment (khalid-davis) [reply README.md:152]: fix

File: README_zh.md (hunk 5 above)
Comment (khalid-davis) [reply README_zh.md:260]: fix

**Comment by khalid-davis** [comment]
@fisherxu

**Pull request merged**